app/controllers/posts_controller.rb

```diff
@@ -54,15 +54,17 @@ def post_params
   def can_edit
     @post = Post.find(params[:id])
     unless @post && current_user && current_user.can_edit?(@post)
-      redirect_to user_post_path(current_user)
+      flash[:edit_not_allowed] =
+        'Post can only be deleted or edited by its author!'
+      redirect_to user_posts_path(current_user)
     end
   end
 
   def check_time!
     if Time.now > @post.created_at + 10.minutes
       flash[:created_at] =
         'Post can only be edited 10 min after it has been created'
-      redirect_to user_posts_path(current_user)
+      redirect_to user_post_path(current_user)
```

Comment:
Shouldn't this be user_posts_path?

Reply:
resolved - also changed the argument to (@post.recipient_id)